build.rs

```diff
@@ -67,9 +67,145 @@
+      "crypto/curve25519/asm/x25519-asm-arm.S",
+      "crypto/ec/asm/ecp_nistz256-armv4.pl",
+      "crypto/modes/asm/ghash-armv4.pl",
+      "crypto/poly1305/asm/poly1305-armv4.pl",
+      "crypto/sha/asm/sha256-armv4.pl",
+      "crypto/sha/asm/sha512-armv4.pl"];
+
+const RING_AARCH64_SRCS: &'static [&'static str] =
+    &["crypto/cpu-aarch64-linux.c",
+      "crypto/bn/asm/armv8-mont.pl",
+      "crypto/chacha/asm/chacha-armv8.pl",
+      "crypto/ec/asm/ecp_nistz256-armv8.pl",
+      "crypto/poly1305/asm/poly1305-armv8.pl",
+      "crypto/sha/asm/sha256-armv8.pl",
+      "crypto/sha/asm/sha512-armv8.pl"];
+
+const RING_TEST_SRCS: &'static [&'static str] =
+    &["crypto/bn/bn_test.cc",
+      "crypto/constant_time_test.c",
+      "crypto/test/bn_test_convert.c",
+      "crypto/test/bn_test_lib.c",
+      "crypto/test/file_test.cc"];
+
+const RING_HEADERS: &'static [&'static str] = &["crypto/poly1305/internal.h",
+                                                "crypto/test/scoped_types.h",
+                                                "crypto/test/rand.h",
+                                                "crypto/curve25519/internal.h",
+                                                "crypto/cipher/internal.h",
+                                                "crypto/bn/rsaz_exp.h",
+                                                "crypto/bn/internal.h",
+                                                "crypto/internal.h",
+                                                "crypto/rsa/internal.h",
+                                                "crypto/modes/internal.h",
+                                                "crypto/ec/ecp_nistz.h",
+                                                "crypto/ec/ecp_nistz384.h",
+                                                "crypto/ec/ecp_nistz256.h",
+                                                "crypto/ec/gfp_internal.h",
+                                                "include/openssl/type_check.h",
+                                                "include/openssl/mem.h",
+                                                "include/openssl/bn.h",
+                                                "include/openssl/opensslconf.\
+                                                 h",
+                                                "include/openssl/arm_arch.h",
+                                                "include/openssl/cpu.h",
+                                                "include/openssl/rsa.h",
+                                                "include/openssl/aes.h",
+                                                "include/openssl/base.h",
+                                                "include/openssl/err.h"];
+
+const RING_TEST_HEADERS: &'static [&'static str] =
+    &["crypto/test/bn_test_lib.h",
+      "crypto/test/file_test.h",
+      "crypto/test/bn_test_util.h"];
+
+const RING_PERL_INCLUDES: &'static [&'static str] = &["crypto/sha/asm/sha-x86_64.pl"];
```

Comment:
`crypto/sha/asm/sha-armv8.pl` is another thing that should be added to this list.